### PR TITLE
Redux toolkit integration

### PR DESCRIPTION
**Pull request by Ash-D23**
Redux Toolkit Setup 
Added Auth using redux
Fetch Posts and Profile Details using Reducer

### CODE REVIEW

**Review by SK02K1 — APPROVED** [review]

File: src/App.js [hunk 1]
```diff
@@ -7,13 +7,15 @@ import Mockman from "mockman-js";
...
+import { useSelector, useDispatch } from 'react-redux'
 
 function App() {
 
   const [mode, setMode] = useLocalStorage("theme", "dark");
 
-  const {user} = useAuthContext()
+  const { user } = useSelector(state => state.user)
+
+  console.log(user)
```
Comment (narharikale) [inline src/App.js]: console.log here

File: src/Pages/Dashboard/Dashboard.jsx [hunk 2]
```diff
@@ -1,11 +1,23 @@
...
 function Dashboard() {
+
+  const { posts } = useSelector(state => state.posts)
+  const dispatch = useDispatch()
+
+  useEffect(() => {
+    dispatch(fetchPosts())
+  }, [])
+
+  console.log(posts)
```
Comment (narharikale) [inline src/Pages/Dashboard/Dashboard.jsx]: here also

**Review by narharikale — APPROVED** [review]
Well structure and clean code.
greate work✨, good to go

File: src/Pages/Dashboard/Dashboard.jsx (hunk 2 above)
Comment (Ash-D23) [reply src/Pages/Dashboard/Dashboard.jsx]: removed

**Pull request merged**